UnitTests/UI/Controls/Test_MasterDetailsView.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Toolkit.Uwp.UI.Controls;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.AppContainer;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UnitTests.UI.Controls
+{
+    [TestClass]
+    public class Test_MasterDetailsView
+    {
+        [TestCategory("MasterDetailsView")]
+        [UITestMethod]
+        public void Test_SelectedIndex_Default()
+        {
+            var items = Enumerable.Range(1, 10).ToArray();
+            var masterDetailsView = new MasterDetailsView();
+            masterDetailsView.ItemsSource = items;
+            Assert.AreEqual(-1, masterDetailsView.SelectedIndex);
+        }
+
+        [TestCategory("MasterDetailsView")]
+        [UITestMethod]
+        public void Test_SelectedItem_Default()
+        {
+            var items = Enumerable.Range(1, 10).ToArray();
+            var masterDetailsView = new MasterDetailsView();
+            masterDetailsView.ItemsSource = items;
+            Assert.IsNull(masterDetailsView.SelectedItem);
+        }
+
+        [TestCategory("MasterDetailsView")]
+        [UITestMethod]
+        public void Test_SelectedIndex_Syncs_SelectedItem()
+        {
+            var items = Enumerable.Range(1, 10).ToArray();
+            var masterDetailsView = new MasterDetailsView();
+            masterDetailsView.ItemsSource = items;
+            masterDetailsView.SelectedIndex = 6;
```

Comment:
>masterDetailsView.SelectedIndex = 6; [](start = 12, length = 36)

You should also add some tests to validate the edge cases (setting -1 in the index, null to the selected item...)

Reply:
Tests added :-)